### DISCUSSION TITLE
part-ii/2019-24/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#36] [Question comments]
Part II 2019 Automata and Formal Languages: Paper 3, Section I, $4 \mathrm{H}$ 

https://questions.tripos.org/part-ii/2019-24/

**Comment by callumhobbis** [comment]
<details>
<summary>Sketch solution</summary>

**(a):** Notes notes notes.

**(b):** Was example sheet 4 Q10(d*). Essentially each production of the form $A\to BC$ adds a nonterminal, and $A\to a$ converts a nonterminal to a terminal. So if $|v|=n$, we need $n-1$ of the first type and $n$ of the second type, so exactly $2n-1$ productions will be done. You can describe an algorithm which applies all possible combinations of $2n-1$ productions, and then checks if $v$ is in the resulting set of words.

**(c):** This is more complicated to explain than it actually is - if you understand CFGs, you can do probably do this.

**(d):** $G$ can't produce $\varepsilon$, so yes!

</details>
